app/ui/src/Components.elm

```diff
@@ -0,0 +1,33 @@
+module Components exposing (..)
+
+import Html exposing (..)
+import Html.Attributes exposing (..)
+import Html.Events exposing (onClick, onInput, onSubmit)
+import Model exposing (Output(..))
+
+
+layout : List (Html msg) -> Html msg
+layout children =
+    div [] children
+
+
+query : (String -> msg) -> msg -> Html msg
+query queryChange submit =
+    Html.form [ onSubmit submit ]
+        [ label [ for "query" ] [ text "Your query: " ]
+        , input [ id "query", onInput queryChange ] []
+        , button [ type_ "submit", onClick submit ] [ text "Update" ]
+        ]
+
+
+output : Output -> Html msg
```

Comment:
@lpil To clarify my sudden change of opinion: I went to remove it and realised I actually need it, otherwise I can't really do what I want the UI to do.